mantidproject/mantid
# Create script to run doxygen

**Description of work**
This PR moves the script used to run Doxygen from a script stored in the Jenkins configuration, into a script within the mantid buildconfig folder. This ensures that the script is now under source control, and is backed-up within the Mantid repository.

It turns out, we have been using a different doxygen version for our PR builds (`1.8.13`) vs our local builds (`1.9.*`). This PR also unifies these versions and fixes any warnings that arise from the doxygen update.

**To test:**
Code review &
Check that this job has passed in the doxygen pipeline:
https://builds.mantidproject.org/job/pull_requests-doxygen/49175/

Fixes #34688 

*This does not require release notes* because **its not a change which is visible to users.**

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
